code/script.js

```diff
@@ -218,108 +220,141 @@ const generateBoard = () => {
   const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
+  
+// Saves value varible -> keep varible
+  let keep 
+  if (category === 'eyes' || category === 'hair') {
+    keep = value === secret[category];
   } else if (category === 'accessories' || category === 'other') {
-
+    keep = secret[category].includes(value); 
   }
+  // Invoking the function filterCharacters with the varible keep
+  filterCharacters(keep);
 }
-
-// It'll filter the characters array and redraw the game board.
+// It´ll filter the characters array and redraw the game board.
+// Alert popup that tells if´s: yes - else: no
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+  
+    if (category === 'accessories') {
+      if (keep) {
+        alert(`Yessss, that person wears ${value}! save all that wears ${value}`);
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
+      } else {
+        alert(`Nooo, that person doesn't wear ${value}! Remove all people who wears ${value}`);
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
+      }
+   }
+    else if (category === 'other') {
+      if (keep) {
+        alert(`Yeeeep! That´s right, the person is a ${value}! Keep all ${value}s here!`);
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
+      } else {
+        alert(`Nope! The person is not a ${value}! Remove all of the ${value}s`);
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
+      }
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
-  }
 
+    else {
+      if (keep) {
+        alert(`Yes, the person have ${value} ${category}! Keep all people that have ${value} ${category}`);
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
+      } else {
+        alert(`Noo! The person do not not have ${value} ${category}. Remove all the ones with ${value} ${category}`)
```

Comment:
Here I found a double negation, which, if it's meant as a clever detail to puzzle us, would have been a fun feature – but I'm guessing it's not supposed to be there :)